src/mempool/mempool.rs

```diff
@@ -488,6 +488,31 @@ impl Mempool {
             .message_router
             .route_fid(fid, self.read_node_mempool.num_shards);
 
+        self.insert_into_shard(shard_id, message.clone(), source.clone())
+            .await;
+
+        // Fname transfers are mirrored to both the sender and receiver shard.
+        if let MempoolMessage::ValidatorMessage(inner_message) = &message {
+            if let Some(fname_transfer) = &inner_message.fname_transfer {
+                let mirror_shard_id = self
+                    .read_node_mempool
+                    .message_router
+                    .route_fid(fname_transfer.from_fid, self.read_node_mempool.num_shards);
+
+                if mirror_shard_id != shard_id {
+                    self.insert_into_shard(mirror_shard_id, message, source)
```

Comment:
hmm, this technically affects consensus. But we might be ok if nodes upgrade at different times since this logic is only hit in the proposer? Other nodes will behave the same regardless of version, as long as the transfer is in the transactions list.

Reply:
Not sure it will affect consensus in practice since it only changes where the message is stored tbh.